index/scorch/persister.go

```diff
@@ -136,15 +140,57 @@ OUTER:
 		case <-w.notifyCh:
 			// woken up, next loop should pick up work
 			continue OUTER
-		case ew := <-s.persisterNotifier:
+		case ew = <-s.persisterNotifier:
 			// if the watchers are already caught up then let them wait,
 			// else let them continue to do the catch up
 			persistWatchers = append(persistWatchers, ew)
-			notifyWatchers()
 		}
 	}
 }
 
+func notifyMergeWatchers(lastPersistedEpoch uint64,
+	persistWatchers []*epochWatcher) []*epochWatcher {
+	var watchersNext []*epochWatcher
+	for _, w := range persistWatchers {
+		if w.epoch < lastPersistedEpoch {
+			close(w.notifyCh)
+		} else {
+			watchersNext = append(watchersNext, w)
+		}
+	}
+	return watchersNext
+}
+
+func (s *Scorch) pausePersisterForMergerCatchUp(lastPersistedEpoch uint64, lastMergedEpoch *uint64,
+	persistWatchers []*epochWatcher) []*epochWatcher {
+
+OUTER:
+	for {
+
+		// first, let the watchers proceed if they lag behind
+		persistWatchers = notifyMergeWatchers(lastPersistedEpoch, persistWatchers)
+
+		// check for slow merger and pause persister until merger catch up
+		if lastPersistedEpoch > *lastMergedEpoch &&
+			lastPersistedEpoch-*lastMergedEpoch > epochDistance {
```

Comment:
since these are uint's, i become a wary about subtraction.  I see you've guarded the overflow/underflow case on line 174, but maybe it can be collapse into a single line, like...

    if lastPersistedEpoch > *lastMergedEpoch + epochDistance {